tools/addrundata/case.go

```diff
@@ -174,22 +162,19 @@ var errNoop = errors.New("already up to date")
 
 // write commits the fixed rundata to the filesystem.
 func (tc *testcase) write(testdir string) error {
-	if tc.pkg == "" {
-		return errors.New("missing test package name")
-	}
 	if tc.fixed == nil {
 		return errors.New("test case was not fixed")
 	}
-	if proto.Equal(tc.existing, tc.fixed) {
+	if proto.Equal(tc.existing, tc.fixed) && !(*checkDeprecated && tc.deprecated) {
```

Comment:
```suggestion
	ok := !tc.deprecated || !*checkDeprecated
	if ok && proto.Equal(tc.existing, tc.fixed) {
```

Reply:
"ok" seemed a little cryptic to me in this context, but I did go with a temporary variable